src/verdant/panel/events/event-date.tsx

```diff
@@ -30,14 +32,22 @@ class NotebookEventDate extends React.Component<NotebookDate_Props> {
           <div className={DATE_LABEL}>
             {Checkpoint.formatDate(this.props.date)}
           </div>
-          <div className={`${DATE_ARROW} ${this.props.isOpen ? "" : "closed"}`}
-               onClick={() => {
-                 if (this.props.isOpen) this.props.close(this.props.date_id);
-                 else this.props.open(this.props.date_id);
-               }}>
+          <div className={DATE_COLLAPSE_HEADER}>
+            <div
+              className={`${DATE_COUNT} ${this.props.isOpen ? "hidden" : ""}`}>
+              ({this.props.events.length})
+            </div>
+            <div
+              className={`${DATE_ARROW} ${this.props.isOpen ? "" : "closed"}`}
+              onClick={() => {
+                if (this.props.isOpen) this.props.close(this.props.date_id);
+                else this.props.open(this.props.date_id);
+              }}>
+            </div>
           </div>
         </div>
-        <div className={`${DATE_GROUP} ${this.props.isOpen ? "" : "hidden"}`}>
+        <div
+          className={`${DATE_GROUP} ${this.props.isOpen ? "" : "hidden"}`}>
           {this.props.events.map((_, index) => {
             let reverse = this.props.events.length - 1 - index;
             return (
```

Comment:
Instead of making props.isOpen toggle a CSS attribute for hide/show with display:none, it may be more efficient to toggle whether the set of elements are rendered at all in the first place. This could be done by taking the "guts" of the DATE_GROUP JSX and separating it out into a function that returns all the normal content if the date group is open and return null if not.

Reply:
Updated, does that look good?